File: src/main/java/com/nextspring/modules/product/ProductEntity.java
package com.nextspring.modules.product;

import java.math.BigDecimal;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.nextspring.modules.category.CategoryEntity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "products")
public class ProductEntity {
	@Id
	private UUID id;
	private String name;
	private BigDecimal price;
	private CategoryEntity category;
}

File: src/main/kotlin/com/nextspring/modules/product/ProductEntity.kt
package com.nextspring.modules.product

import com.nextspring.modules.category.CategoryEntity
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigDecimal
import java.util.*

@Document(collection = "products")
data class ProductEntity(
    @Id
    var id: UUID? = null,
    var name: String? = null,
    var price: BigDecimal? = null,
    var category: CategoryEntity? = null,
)

data class ProductInput(val name: String, val categoryId: UUID, val price: BigDecimal)
